receiver.go

```diff
@@ -155,7 +155,7 @@ func (r *Receiver) AcceptMessage(ctx context.Context, msg *Message) error {
 // If the context's deadline expires or is cancelled before the operation
 // completes, the message's disposition is in an unknown state.
 func (r *Receiver) RejectMessage(ctx context.Context, msg *Message, e *Error) error {
-	return r.messageDisposition(ctx, msg, &encoding.StateRejected{Error: e})
+	return msg.rcv.messageDisposition(ctx, msg, &encoding.StateRejected{Error: e})
```

Comment:
I think we need a nil check here for 'rcv' (similar in other spots)

Reply:
(ie, if the user settles twice we don't want a panic)

Reply:
It's legal to invoke methods on a `nil` receiver. You just can't dereference said receiver in the method else it will panic.

Reply:
Once the message has been settled, attempting to settling it again is a no-op and will exit early.

Reply:
Do you mind adding a comment here that that's expected? I know it's possible to use nil receivers but I don't often expect them.

https://github.com/Azure/go-amqp/blob/f38b50152cfd7fcdf040d9e3b2781618b6689e0a/receiver.go#L273

Reply:
Good idea, done.